api/src/test/java/org/openmrs/notification/MessageServiceTest.java

```diff
@@ -113,4 +119,68 @@ public void sendMessage_shouldSendMessage() throws MessageException {
 		}
 	}
 	
+	@Test
+	public void setMessagePreparator_shouldSetMessagePreparator() throws MessageException {
+		MessagePreparator messagePreparator = new VelocityMessagePreparator();
+		MessageService messageService = Context.getMessageService();
+		messageService.setMessagePreparator(messagePreparator);
+		assertEquals(messagePreparator, messageService.getMessagePreparator());
+	}
+
+	@Test
+	public void setMessageSender_shouldSetMessageSender() throws MessageException {
+		MessageSender messageSender = new MailMessageSender();
+		MessageService messageService = Context.getMessageService();
+		messageService.setMessageSender(messageSender);
+		assertEquals(messageSender, messageService.getMessageSender());
+	}
+	
+	@Test(expected = MessageException.class)
+	public void sendMessage_shouldFailIfMessageIsNull() throws MessageException {
+		MessageService messageService = Context.getMessageService();
+		messageService.sendMessage(null);
+	}
+
+	@Test
+	public void createMessage_shouldSetAppopriateValuesForParameters() throws MessageException {
+		MessageService messageService = Context.getMessageService();
+		String subject = "Message Subject";
+		String recipients = "Message Recipients";
+		String sender = "Message Sender";
+		String content = "Message Content";
+		String attachment = "Message Attachment";
+		String attachmentFileName = "Message Attachment File name";
+		String attachmentContentType = "Message attachment content type";
+		Message message = messageService.createMessage(recipients,sender,subject,content,attachment,attachmentContentType,attachmentFileName);
+		
+		assertEquals(message.getSubject(),subject);
+		assertEquals(message.getRecipients(),recipients);
+		assertEquals(message.getSender(),sender);
+		assertEquals(message.getContent(),content);
+		assertEquals(message.getAttachment(),attachment);
+		assertEquals(message.getAttachmentContentType(),attachmentContentType);
+		assertEquals(message.getAttachmentFileName(),attachmentFileName);
+	}
+	
+	@Test
+	public void sendMessage_shouldSetRecipientIfAddressIsNotNull() throws MessageException {
+		String recipient1 = "address1";
+		String recipient2 = "address2";
+		Context.getAuthenticatedUser().setUserProperty(OpenmrsConstants.USER_PROPERTY_NOTIFICATION_ADDRESS, recipient2);
+		MessageSender messageSender = new MailMessageSender();
+		Context.getMessageService().setMessageSender(messageSender);
+		Message messageObj = new Message(1,recipient1,"sender","subject","content");
+		try {
+			Context.getMessageService().sendMessage(messageObj, Context.getAuthenticatedUser());
+		}
+		catch (MessageException e) {
+		//So that this test doesn't fail just because the user isn't running an SMTP server.
+		if (!e.getMessage().contains(NO_SMTP_SERVER_ERROR)) {
+			e.printStackTrace();
+			fail();
+		}
+	}
+		assertEquals(messageObj.getRecipients(),recipient1 + "," + recipient2);
```

Comment:
What happen with the messageObj if the Exception is thrown . Will be the Recipients null?

Reply:
The Exception throw was redundant, so I removed it

Reply:
The Exception throw was not necessary, so I removed it.